### ISSUE TITLE
Download of payment reports is (for me) not working any more.

### ISSUE THREAD

**Issue by meyer-solutions** [nicolasgomollon/AppSales-Mobile#67]
In my special case the provider selection stops working someday near the end of 2023 which leads to problems in downloading payment reports.
After some investigation of the Apple developer website I found that adding the "X-Request-With" header solves this issue.

`[authSessionRequest setValue:@"olympus-ui" forHTTPHeaderField:kITCRXRequestedWithKey];`

**Comment by ddaddy** [comment]
Is this for the daily reports, or the monthly payments in the Payments section?
What do you mean 'provider selection'?

**Comment by meyer-solutions** [comment]
@ddaddy I mean the Payments section, which looks like a calendar. (**not** Sales and Trends)

'provider selection' means the method name `changeToProvider` calling `/olympus/v1/session`

**Comment by ddaddy** [comment]
Ah ok. Mine have been working, although I have 2 accounts and have to enable/disable to do one at a time as the 2FA doesn't work with 2 at a time.
I'll add your line to my current pull request which has several fixes and improvements.

I don't know what's happened to @nicolasgomollon I'm tempted to start my own branch of this is the PR doesn't get approved soon.

**Comment by meyer-solutions** [comment]
Thank you - I also already using my own branch. I have around 10 accounts with 2FA enabled.

**Comment by ddaddy** [comment]
Oh wow. Does it manage to connect to them all?
I disable payments until after a payment has been made as it adds overhead to the daily requests.

**Comment by meyer-solutions** [comment]
No, I'm using AppSales only for my personal account and not for the customers. But Apple is choosing the first provider after the login randomly and AppSales has to change it every time after login.

**Comment by ddaddy** [comment]
I've added that to my PR #65 
It seems with multiple accounts, it starts logging into the 2nd account before the 1st has completed the Payments login. If I disable my 2nd account I can get the first to download its payments.

When I get time I might add either a Payments toggle per account, or a Disable toggle per account to be able to temporarily halt.

You should check out my PR, I love the ability to filter and show how many New Subscriptions I get each day.

**Comment by ddaddy** [comment]
Done it :)
Added the ability to disable downloading per account.

**Comment by pupasani** [comment]
I have a single account, payments download has stopped working since the beginning of 2026

**Comment by ddaddy** [comment]
I fixed this is my branch here: https://github.com/ddaddy/AppSales-Mobile/tree/subscriptions

I don’t know why my pull requests get ignored, I think @nicolasgomollon no longer pays attention to this repo.

**Comment by nicolasgomollon** [comment]
Fixed in #65 which was just merged.
